src/software/thunderscope/robot_diagnostics/robot_info.py

```diff
@@ -0,0 +1,199 @@
+import pyqtgraph as pg
+from pyqtgraph.Qt import QtCore, QtGui
+from pyqtgraph.Qt.QtWidgets import *
+from software.py_constants import *
+from proto.import_all_protos import *
+import software.thunderscope.common.common_widgets as common_widgets
+from software.thunderscope.constants import *
+
+
+class RobotInfo(QWidget):
+
+    toggle_one_connection_signal = QtCore.pyqtSignal(int, int)
+
+    def __init__(self, robot_id, load_fullsystem, control_mode_signal):
+        """
+        Initialize a single robot's info widget
+
+        :param robot_id: id of robot whose info is being displayed
+        :param load_fullsystem: whether fullsystem is loaded currently
+                                Shows / Hides AI option from control mode menu accordingly
+        :param control_mode_signal: signal that should be emitted when a robot changes control mode
+        """
+
+        super().__init__()
+
+        self.robot_id = robot_id
+        self.control_mode_signal = control_mode_signal
+
+        # when this robot has a battery warning, this is set to True
+        # which prevents spamming the same battery warning
+        # set back to False if battery is back above warning level
+        self.battery_warning_disabled = False
+
+        self.layout = QHBoxLayout()
+
+        self.status_layout = QVBoxLayout()
+
+        # Offsets the minimum of the battery bar from the minimum ideal voltage
+        # Allows battery % to go below the minimum ideal level
+        BATTERY_MIN_OFFSET = 3
```

Comment:
constants should be consolidated in the class headers itself (so move this to line 11)

Reply:
done